### PR TITLE
Add SE-ResNext Model

### PR DESCRIPTION
**Pull request by hetong007**
1. Add SE-ResNet, SE-ResNext, SENet
2. Add label smoothing for SENet training

### CODE REVIEW

**Comment by mli** [comment]
Job PR-104-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-104/3/index.html

File: gluoncv/model_zoo/se_resnet.py [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+           'get_se_resnet']
+
+import os
+from mxnet import cpu
+from mxnet.gluon import nn
+from mxnet.gluon.block import HybridBlock
+
+# Helpers
+def _conv3x3(channels, stride, in_channels):
+    return nn.Conv2D(channels, kernel_size=3, strides=stride, padding=1,
```
[the remark this answers is not in the document]
Comment (hetong007) [reply gluoncv/model_zoo/se_resnet.py]: ResNext is defined in gluoncv, thus I currently adding one parameter to control the use of SE layers. For the SENet-154, there's a senet.py file specifically for that.

On the other hand, ResNet is defined in main mxnet repo, thus it's less flexible to modify code there.
